src/training/ea/lipizzaner_trainer.py

```diff
@@ -137,6 +140,74 @@ def optimize_generator_mixture(gans):
         # TODO: implement
         pass
 
+    #
+    # MUTATION
+    #
+
+    def _mutate(
+        self,
+        attackers: Population,
+        defenders: Population,
+        input_data,
+        input_labels=None,
+    ):
+        return self.step(attackers, defenders, input_data, labels=input_labels)
+
+    def step(self, attacker, defender, input_data, labels=None):
+        self.mutate_hyperparams(attacker)
+        return self.update_genomes(attacker, defender, input_data, labels=labels)
+
+    def mutate_hyperparams(self, population):
```

Comment:
This could be a method inside `Population`, receiving all the extra parameters of course